### PR TITLE
Mcgov/dpdk send receive refactor

### PR DESCRIPTION
**Pull request by mcgov**
Cleaning up some logic which has been tripping send/receive testing up.
- refactor send/receive
   - ensure receiver starts first
   - simplify multiple queue test logic to allow use of shared logic
   - simplify queue and core selection to fix issues across variable threads/core/numa configurations.
   - rename variables to make logical core vs physical core model in DPDK EAL explicit.

### CODE REVIEW

File: lisa/tools/timeout.py [hunk 1]
```diff
@@ -12,20 +12,36 @@ def can_install(self) -> bool:
...
+        timeout: int,
+        signal: int = SIGTERM,
+        kill_timeout: int = 0,
+        delay_start: int = 0,
+    ) -> Process:
+        # timeout [OPTION] DURATION COMMAND [ARG]...
         params = f"-s {signal} --preserve-status {timeout} {command}"
         if kill_timeout:
             params = f"--kill-after {kill_timeout} " + params
-        command_timeout = timeout
```
Comment (squirrelsc) [inline lisa/tools/timeout.py:21]: It looks this logic is missed. It should be in run_with_timeout.

Comment (mcgov) [reply lisa/tools/timeout.py:21]: apologies, merge issue needs to be resolved. I'll abandon this and re-open when it's ready

**Comment by mcgov** [comment]
Open for review, merge issues resolved and testing is in progress.

File: lisa/tools/timeout.py (hunk 1 above)
Comment (squirrelsc) [reply lisa/tools/timeout.py:21]: Is it added back?

**Comment by mcgov** [comment]
Something weird is happening w github, going to open a new PR to refresh the contents.

**Pull request closed without merging**